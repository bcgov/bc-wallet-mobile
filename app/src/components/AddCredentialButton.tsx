import { useTheme, testIdWithKey } from 'aries-bifold'
import React, { useCallback } from 'react'
import { useTranslation } from 'react-i18next'
import { StyleSheet, TouchableOpacity, DeviceEventEmitter } from 'react-native'
import Icon from 'react-native-vector-icons/MaterialCommunityIcons'

import { BCWalletEventTypes } from '../events/eventTypes'

const AddCredentialButton: React.FC = () => {
  const { t } = useTranslation()
  const { ColorPallet } = useTheme()

  const activateSlider = useCallback(() => {
    DeviceEventEmitter.emit(BCWalletEventTypes.ADD_CREDENTIAL_PRESSED, true)
  }, [])

  const styles = StyleSheet.create({
    button: {
      paddingHorizontal: 16,
    },
  })
  return (
    <TouchableOpacity
      accessible={true}
      accessibilityLabel={t('Credentials.AddCredential')}
      testID={testIdWithKey('AddCredential')}
      style={styles.button}
      onPress={activateSlider}
    >
      <Icon name="plus-circle-outline" size={24} color={ColorPallet.grayscale.white}></Icon>
    </TouchableOpacity>
  )
}

export default AddCredentialButton
