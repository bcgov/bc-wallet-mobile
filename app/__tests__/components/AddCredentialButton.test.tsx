import { render } from '@testing-library/react-native'
import React from 'react'

import AddCredentialButton from '../../src/components/AddCredentialButton'

describe('AddCredentialButton Component', () => {
  test('renders correctly', () => {
    const tree = render(<AddCredentialButton />)
    expect(tree).toMatchSnapshot()
  })
})
