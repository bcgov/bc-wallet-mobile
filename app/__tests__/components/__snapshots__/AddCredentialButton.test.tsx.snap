// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddCredentialButton Component renders correctly 1`] = `
<View
  accessibilityLabel="Credentials.AddCredential"
  accessibilityRole="button"
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  hitSlop={
    {
      "bottom": 44,
      "left": 44,
      "right": 44,
      "top": 44,
    }
  }
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "opacity": 1,
    }
  }
  testID="com.ariesbifold:id/AddCredential"
>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "marginLeft": 0,
        "marginRight": 15,
        "minHeight": 26,
        "minWidth": 26,
      }
    }
  >
    <Text
      allowFontScaling={false}
      selectable={false}
      style={
        [
          {
            "color": "#FFFFFF",
            "fontSize": 26,
          },
          undefined,
          {
            "fontFamily": "Material Design Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          {},
        ]
      }
    >
      Û∞êô
    </Text>
  </View>
</View>
`;
